### DISCUSSION TITLE
Suggested new defaults / configuration properties

### DISCUSSION THREAD

**Discussion by aru-rcollina** [scalar/scalar#2648] [Ideas]
Hello,
Love this project!

I believe it could be useful to either default Body to None or make it configurable when testing DELETE methods.

![image](https://github.com/user-attachments/assets/28d6a7fb-81aa-4e1f-b514-7546b0123f97)

I'd also like to be reminded that the `{value}` route param is actually a mandatory field before issuing the request, instead of defaulting it to, well, `{value}`. :)

**Comment by hanspagel** [comment]
Ah, we’re not adding it as `{value}`, but replace it. With the configuration in your screenshot, we’d replace it with an empty value. Which should always be allowed (to test how an API handles bad input). I think we could add a different color to the variable, so it’s clear it’s not just text, but we “know” it’s a variable and replace it. Maybe we can come up with more improvements to improve this.

**Reply by aru-rcollina** [reply]
Hello @hanspagel, I believe I may have prevented notifications from reaching you by replying [below](https://github.com/scalar/scalar/discussions/2648#discussioncomment-10275882) instead of replying to this comment.

**Comment by aru-rcollina** [comment]
Hello @hanspagel, thanks for your reply.
Unless there's something wrong with my setup - might as well be - I'm observing `{value}` as a default when no value is defined by the user.

![image](https://github.com/user-attachments/assets/7abbfb82-d2cf-4e6f-ba0b-ea8f48cf52f8)

![image](https://github.com/user-attachments/assets/0031c5a9-0e44-4e3b-bada-4a66e38f5119)

![image](https://github.com/user-attachments/assets/f37bb5cf-2829-40b1-a134-a7df80b7013a)

I thought I'd test via the demo site as well and there seems to be no replacement, either:

![image](https://github.com/user-attachments/assets/82404071-1d5f-41a7-b040-0b6fd33f4299)

This is the OpenAPI document I'm using, generated via Swagger (.NET) 

```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "Title",
    "description": "Description",
    "contact": {
      "name": "Name",
      "url": "https://www.example.com"
    },
    "version": "2.0.0"
  },
  "paths": {
    "/api/v2/examples/{value}": {
      "delete": {
        "tags": [
          "Examples"
        ],
        "parameters": [
          {
            "name": "value",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/Payload"
                },
                "examples": {
                  "example-1": {
                    "value": {
                      "value": "example-1"
                    }
                  }
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Payload"
                },
                "examples": {
                  "example-1": {
                    "value": {
                      "value": "example-1"
                    }
                  }
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/Payload"
                },
                "examples": {
                  "example-1": {
                    "value": {
                      "value": "example-1"
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "Payload": {
        "type": "object",
        "properties": {
          "value": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      }
    }
  }
}
```

Anything I should be looking out for?
